### ISSUE TITLE
Add new option to all stock list dropdowns to use current inventory at location

### ISSUE THREAD

**Issue by jakkarth** [openboxes/openboxes#1029]
Using 0.8.6

Actual behavior:

Functionality does not exist.

Expected behavior:

Whether creating a requisition, stock movement, or shipment (#1018), we have the option of starting from a stock list: a list of products with a default quantity and uom. Those defined stock lists are presented in a dropdown at the start of each workflow, and a blank "none" option can be selected to start with an empty list if desired. For this ticket, we're requested an additional item be added to that list: current location inventory. Selecting this would prepopulate the list for the action not from a predefined stocklist, but would be a list of every product currently in inventory at the origin location, with the default quantity being the current inventory counts. Perhaps a clearer way to describe it would be through a use case: I am closing my dispensary down, and I need to ship the entire inventory back to the depot; rather than define a stock list with every single item in stock on it, I could just select this option and have the current inventory populate the shipment.

Steps to reproduce:

1. Select Create stock requisition, Create outbound movement, or Create outbound shipment
2. In the stocklist dropdown, select the "all inventory at origin location" option
3. On subsequent steps, the product list and quantities will be based on the inventory currently available at the origin location

**Comment by jmiranda** [comment]
@jakkarth So I've looked into this a bit and it's less straightforward than I was hoping. I was wondering how you would feel if you could export your current inventory from one location in a format that would allow you to import all items into the stock movement on page 2 of the workflow.

**Comment by jakkarth** [comment]
That sounds fine.

**Comment by jmiranda** [comment]
@jakkarth Are you using lot / serial numbers, expiration dates, and/or bin locations in any of these locations?

**Comment by jmiranda** [comment]
So the issue I'm running into right now is that the import format for an outbound stock movement only supports product and quantity. You need to go through the picking process to select the actual lots. This *should* happen automatically, but it might be a bit weird / jarring.

**Comment by jakkarth** [comment]
I think I can deal with that.On 22 Oct 2019 16:40, Justin Miranda <notifications@github.com> wrote:So the issue I'm running into right now is that the import format for an outbound stock movement only supports product and quantity. You need to go through the picking process to select the actual lots. This should happen automatically, but it might be a bit weird / jarring.

**Comment by jmiranda** [comment]
https://drive.google.com/file/d/1yzwathlXnNBZUS7MNT0-Bob4IQmZvW24/view

**Comment by jmiranda** [comment]
Towards the tail end of the video you'll hear me say that Manage Inventory is not showing up in the menu. That's actually not a bug. The current location doesn't support Adjust Stock activity type so the menu item does not render.

**Comment by jmiranda** [comment]
Let me know if the feature demo'd in the video looks ok to you. We can work on implementing it as a Return Stock feature which would allow you to select the items you want to return and would create a mirrored transaction (transfer out / transfer in) instead of requiring you to use the stock movement feature. Also, we have plans to improve the Import feature in the Outbound Stock Movement workflow to allow users to import data for all pages (requested, revised, and picked items) all at once.

**Comment by jakkarth** [comment]
Watched through the video and it looks like that would be sufficient in the short term. I grabbed the latest daily build from bamboo (http://bamboo.pih-emr.org:8085/browse/OPENBOXES-DRB-57/artifact) but it doesn't have the download as stock movement template option on the bin location report. Is there a different build I should try?

**Comment by jmiranda** [comment]
It's in a separate feature branch that hasn't been PR'd/merged into develop yet. I was just waiting to hear if this was ok. I'll push it to develop tonight.

**Comment by jmiranda** [comment]
The latest daily snapshot is available http://bamboo.pih-emr.org:8085/browse/OPENBOXES-DSB-336

**Comment by jakkarth** [comment]
Yeah, that seems to work. I exported it from the bin report and imported it on the "Add items" step of an outbound stock movement without modification. It combines lots together, but that's ok for my use case.

**Comment by jmiranda** [comment]
That's one of the limitations at the moment since outbound stock movements originate as "requests" they need to be made against the product rather than the inventory item. However, the auto-picking process should prevent you from having to manually pick the individual inventory items. 

The other way we could do it is as an inbound stock movement. However, I don't think the inbound stock movement creates a debit from the origin location so you'd have to manually clear out the inventory in the origin.

**Comment by jmiranda** [comment]
On that note, would you prefer to see a "Return Stock" action that allows you to run the bin location report, select items to return, select the "return to" destination and then create a mirrored debit / credit transaction representing the transfer out / transfer in of the returned stock?

**Comment by jakkarth** [comment]
I think for the moment I'm comfortable using the existing stock movement system. It makes intuitive sense to me and seems to have the basic features I need. I'll report back after the event is over and I've used it all the way through the cycle and try to provide better feedback at that point.
